lib/Migration/Version040100Date20240611165300.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2024 Jonas <jonas@freesources.org>
+ *
+ * @author Jonas <jonas@freesources.org>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Text\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\Types;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version040100Date20240611165300 extends SimpleMigrationStep {
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		if ($schema->hasTable('text_steps')) {
+			$table = $schema->getTable('text_steps');
+			if (!$table->hasColumn('timestamp')) {
+				$table->addColumn('timestamp', Types::BIGINT, [
+					'notnull' => true,
+					'length' => 6,
+					'default' => 'CURRENT_TIMESTAMP',
```

Comment:
Do you know if this works on all our possible database backends? Or does doctrine maybe handle that?

Reply:
Good point. I looked it up now and MySQL, MariaDB, PostgreSQL, SQLite and Oracle all support `DEFAULT CURRENT_TIMESTAMP` since a long time. But I also searched our code and only the health app uses it. Circles [used it](https://github.com/nextcloud/circles/pull/14) for a short time but [migrated away shortly afterwards](https://github.com/nextcloud/circles/pull/55) due to [a bugreport](https://github.com/nextcloud/circles/issues/54).

Reply:
Changed now to `DEFAULT 0` and setting the timestamp when inserting the step.